core/plugins/org.polarsys.capella.core.semantic.queries/src/org/polarsys/capella/core/semantic/queries/sirius/diagram/AbstractModelElementRelatedRepresentationQuery.java

```diff
@@ -32,13 +36,46 @@
     if (!(object_p instanceof EObject)) {
       return result;
     }
-    result.addAll(RepresentationHelper.getAllRepresentationDescriptorsWhereSemanticElementIsDisplayed((EObject) object_p, filterRepresentationDescription()));
+    Collection<Object> unfilteredRepresentationDescs = new ArrayList<>();
+    if (QueryResultCache.getInstance().get(object_p) == null) {
```

Comment:
Main concern, an IQuery can be used outside the context of the Semantic Browser view (in Mass visualization views for instance).

Can you move the cache looking outside the IQuery but rather in the AbstractContentProvider of the semantic browser view.
Mostly here: 
https://github.com/eclipse/capella/blob/409a90c4a2ed6c9c355badee0d068c32ec47f195/common/plugins/org.polarsys.capella.common.ui.toolkit.browser/src/org/polarsys/capella/common/ui/toolkit/browser/content/provider/impl/AbstractContentProvider.java#L193


and a clear() cache when inputChanged shall do the job done in the refresh below.

Reply:
I agree that this would have been a good option, I'm keeping the idea, it might be useful depending on the feedback on the Sirius/Collab reviews.